client.go

```diff
@@ -192,6 +199,10 @@ func (c *Client) exchange(m *Msg, a string) (r *Msg, rtt time.Duration, err erro
 	// Otherwise use the client's configured UDP size.
 	if opt == nil && c.UDPSize >= MinMsgSize {
 		co.UDPSize = c.UDPSize
+		// In this case also set the the EDNS0 bufsize to allow for larger responses.
+		if c.SetUDPSize {
+			m.SetEdns0(c.UDPSize, true /* dnssec */)
```

Comment:
[ Quoting <notifications@github.com> in "Re: [dns] Udp buffer 2 (#322)..." ]
>> @@ -192,6 +199,10 @@ func (c *Client) exchange(m *Msg, a string) (r *Msg, rtt time.Duration, err erro
>>  	// Otherwise use the client's configured UDP size.
>>  	if opt == nil && c.UDPSize >= MinMsgSize {
>>  		co.UDPSize = c.UDPSize
>> +		// In this case also set the the EDNS0 bufsize to allow for larger responses.
>> +		if c.SetUDPSize {
>> +			m.SetEdns0(c.UDPSize, true /* dnssec */)
>
>I can't decide if using constants or two bools in the struct is better (SetUDPSize, EnableDNSSEC).  The nice part about using struct members is that the code is just
>```
>if c.SetUDPSize {
>  m.SetEDNS0(c.UDPSize, c.EnableDNSSEC)
>}
>```

That is true, but EnableDNSSEC *also* implies adding a EDNS0 OPT RR, regardless
of the UDPSize, which if the latter isn't set should also have a default of
4k...? But, maybe two bools is better.

>Perhaps SetUDPSize is the wrong bool name for the client though?  Is it really SetEDNS0?

Yep. I think SetEDNS0 is better, to be inline with the rest of the lib, it has
to be SetEdns0 though.

>At some point the code might need to handle different EDNS versions as well, so maybe an EDNS struct that has {UDPSize, DNSSEC, Version, ...future options...} is a better method to abstract the EDNS options?  Then m.SetEDNS (note not m.SetEDNS0) could take an options struct instead of an argument list.  If this route is selected then the Client struct should have an EDNSOptions member that is passed around instead.

*That* is a valid point, but I am

a) not convinced this is going to happen any time soon (could be mistaken
though)
b) warrents the complications it gives to solving the current problem (.i.e
people sending msgs with bufsizes that are too small).

>I haven't placed much thought into what other EDNS options exist to know if this is a valid forward looking solution.

I *do* like the Config/Option struct to pass options around though...